profet/Cleaver.py

```diff
@@ -0,0 +1,185 @@
+        signal_peptides = []
+
+        # Find the signal peptide starting and end positions
+        for child in root:
+            for grandchild in child:
+                if grandchild.attrib.get("type") == "signal peptide":
+                    if (
+                        grandchild[0][0].attrib.get("position")
+                        and grandchild[0][1].attrib.get("position") is not None
+                    ):
+                        start_position = int(
+                            str(grandchild[0][0].attrib.get("position"))
+                        )
+                        end_position = int(
+                            str(grandchild[0][1].attrib.get("position"))
+                        )
+                        signal_peptides.append((start_position, end_position))
+                    else:
+                        print(uniprot_id + "has no signal peptide")
+                else:
+                    None
+        return signal_peptides
+
+    def remove_signal_peptide_pdb(
+        self,
+        pdb_file: str,
+        signal_peptides: list,
+        output_filename: str,
+    ):
+        """
+        Load the pdb file and deletes the signal peptides. Save as new cleaved pdb file.
+
+        Args:
+            pdb_file: Path to the pdb input file
+            signal_peptides: List of signal peptide start and ending positions pairs
+            output_filename: Name of the cleaved protein file you want to write to
+        Returns:
+            None
+        """
+
+        with open(pdb_file, "r") as input_pdb:
+            with open(output_filename, "w") as output_pdb:
+                for line in input_pdb:
+                    if line.startswith("ATOM"):
+                        residue_position = int(
+                            line[22:26].strip()
+                        )  # 22-26 to check for amino acid number/ residue number
+                        for start_position, end_position in signal_peptides:
+                            if (
+                                start_position
+                                <= residue_position
+                                <= end_position
+                            ):
+                                break
+                        else:
+                            output_pdb.write(line)
+                    else:
+                        output_pdb.write(line)
+
+    def remove_signal_peptide_cif(
+        self,
+        input_file: str,
+        signal_peptides: list,
+        output_filename: str,
+    ):
+        """
+        Load the cif file and deletes the signal peptides. Save as new cleaved cif file.
+
+        Args:
+            input_file: Path to the cif input file
+            signal_peptides: List of signal peptide start end ending positions pairs
+            output_filename: Name of the cleaved protein file you want to write to
+        Returns:
+            None
+        """
+        input_file = "/home/akv16273/.cache/pdb/p61316.pdb"
```

Comment:
Hi @NikJur . Did you mean to override the input_file variable here?

Reply:
Oh goodness, no. Good spot!